lib/smart_proxy_openscap/openscap_lib.rb

```diff
@@ -13,10 +13,35 @@
 require 'json'
 require 'proxy/error'
 require 'proxy/request'
+require 'smart_proxy_openscap/openscap_exception'
 
 module Proxy::OpenSCAP
   extend ::Proxy::Log
 
+  def self.get_policy_content(policy_id)
+    policy_store_dir = File.join(Proxy::OpenSCAP::Plugin.settings.contentdir, policy_id.to_s)
+    policy_scap_file = File.join(policy_store_dir, "#{policy_id}_scap_content.xml")
+    FileUtils.mkdir_p(policy_store_dir) # will fail silently if exists
+
+    return File.open(policy_scap_file, 'rb').read if File.file?(policy_scap_file) && !File.zero?(policy_scap_file)
+
+    lock = Proxy::HttpDownloads.try_locking(policy_scap_file)
+
+    response = fetch_scap_content_xml(policy_id, policy_scap_file)
```

Comment:
I would move it to another method for clarity, it took me a minute to find out which part is reading from cache vs fetching again